### PR TITLE
FC-0001: Remove EdxRestApiClient usage in taxonomy-connector

### PR DESCRIPTION
**Pull request by ruzniaievdm**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [ ] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ruzniaievdm!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Review by dianakhuang — APPROVED** [review]
This looks good to me! Going to wait for the owning team to see if they can give a review as well.

**Review by saleem-latif — APPROVED** [review]
Looks Good 👍, I also ran did a test run locally and new client is able to successfully connect to the EMSI.

**Comment by saleem-latif** [comment]
Just make sure to update the [version number](https://github.com/openedx/taxonomy-connector/blob/master/taxonomy/__init__.py#L18) and add a [CHANGELOG](https://github.com/openedx/taxonomy-connector/blob/master/CHANGELOG.rst) entry before merging.

**Comment by ruzniaievdm** [comment]
@saleem-latif already done!

File: taxonomy/emsi/client.py
Comment (sameenfatima78) [inline taxonomy/emsi/client.py:146]: nit: sending the variable "path" as is would to this method would look a lot more neater.

File: taxonomy/emsi/client.py
Comment (sameenfatima78) [inline taxonomy/emsi/client.py:230]: same feedback as above.

**Review by sameenfatima78 — APPROVED** [review]
Looks good. Just a small comment

**Comment by ruzniaievdm** [comment]
@sameenfatima78 Thanks for your feedback! I think that adding additional variables does not make sense, as this will not simplify the debugging process. If you insist i can update the code by means of adding the "path".

**Comment by dianakhuang** [comment]
@ruzniaievdm I'm going to merge this PR now. Nits are usually optional, so I think if you would prefer to keep the code as-is, then we don't need to implement them.

**Comment by openedx-webhooks** [comment]
@ruzniaievdm 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/taxonomy-connector&entry.867055334=https://github.com/openedx/taxonomy-connector/pull/93&entry.1484655318=https://github.com/ruzniaievdm&entry.752974735=2022-07-14+07:45&entry.1917517419=2022-07-18+13:44&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
